src/components/Calculator.js

```diff
@@ -1,53 +1,74 @@
 /* eslint-disable no-unused-vars */
+/* eslint-disable max-len */
 import React from 'react';
 import Buttons from './Buttons.js';
 import Input from './Input.js';
+import calculate from '../logic/calculate.js';
 
 class Calculator extends React.Component {
   constructor(props) {
     super(props);
 
     this.state = {
-      input: 0,
+      total: null,
+      next: null,
+      operation: null,
     };
+    this.addToInput = this.addToInput.bind(this);
+  }
+
+  addToInput = (val) => {
+    calculate(this.state, val);
+    const { next, total, operation } = calculate(this.state, val);
+    if (next !== undefined) {
+      this.setState({ next });
+    }
+
+    if (total !== undefined) {
+      this.setState({ total });
+    }
+
+    if (operation !== undefined) {
+      this.setState({ operation });
```

Comment:
-  [ ] I cannot see division in your project. You have duplicated addition rule. Please work on it.